waddrmgr/manager.go

```diff
@@ -1470,6 +1485,188 @@ func (m *Manager) LastInternalAddress(account uint32) (ManagedAddress, error) {
 	return acctInfo.lastInternalAddr, nil
 }
 
+// NewAccount creates and returns a new account stored in the manager based
+// on the given account name.  If an account with the same name already exists,
+// ErrDuplicateAccount will be returned.  Since creating a new account requires
+// access to the cointype keys (from which extended account keys are derived),
+// it requires the manager to be unlocked.
+func (m *Manager) NewAccount(name string) (uint32, error) {
```

Comment:
Yup, adding the mutex lock and checks as described. We were calling `LookupAccount` with the mutex lock held so I've added a unexpected version `lookupAccount` to make it usable.